src/function/debounce.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { debounce } from './debounce';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+describe('debounce', () => {
+  beforeEach(() => {
+    vi.useFakeTimers();
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
```

Comment:
Let's add a test to make sure that the function is called with the given arguments.

```ts
debouncedFn = debounce(fn)
debouncedFn(1,2,'3');
dbouncedFn.toBeCalledWith(...)
```

Reply:
Yep. My bad, I meant the internal function.